### ISSUE TITLE
NET-1848 Fix S2952 FN: Auto properties are not being covered

### ISSUE THREAD

**Issue by Tim-Pohlmann** [SonarSource/sonar-dotnet#6770]
### Description

Rule [S2952](https://sonarsource.github.io/rspec/#/rspec/S2952/csharp) states:
> Move disposing of **members** to some other method, and you risk resource leaks.

However, the rule only triggers on fields. It should also trigger on auto properties that are non-`static` and `private`/`protected`.

### Repro steps

```CS
public class PropertiesFN : IDisposable
{
    private Stream Stream { get; }

    public void CleanUp()
    {
        Stream.Dispose();  // False Negative
    }

    public void Dispose()
    {
    }
}
```

### Related information

Originally discussed here: https://github.com/SonarSource/sonar-dotnet/pull/6230#discussion_r1108481888

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1848](https://sonarsource.atlassian.net/browse/NET-1848)

[NET-1848]: https://sonarsource.atlassian.net/browse/NET-1848?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by pavel-mikula-sonarsource** [comment]
Moved
